src/io/ODF.cpp

```diff
@@ -45,431 +49,363 @@ void ODF::init(ParticleContainer* particleContainer, DomainDecompBase* domainDec
-		this->_molID[i] = 0;
-		this->_r1[0][i] = 0.;
-		this->_r1[1][i] = 0.;
-		this->_r1[2][i] = 0.;
-		this->_Q1[0][i] = 0.;
-		this->_Q1[1][i] = 0.;
-		this->_Q1[2][i] = 0.;
-		this->_Q1[3][i] = 0.;
-		this->_doODF[i] = 0;
-		this->_upVec1[0][i] = 0.;
-		this->_upVec1[1][i] = 0.;
-		this->_upVec1[2][i] = 0.;
-	}
+	fillZero(_ODF11);
+	fillZero(_ODF12);
+	fillZero(_ODF21);
+	fillZero(_ODF22);
+
+	std::for_each(_threadLocalODF11.begin(), _threadLocalODF11.end(), fillZero);
+	std::for_each(_threadLocalODF12.begin(), _threadLocalODF12.end(), fillZero);
+	std::for_each(_threadLocalODF21.begin(), _threadLocalODF21.end(), fillZero);
+	std::for_each(_threadLocalODF22.begin(), _threadLocalODF22.end(), fillZero);
 }
 
-void ODF::collectTempParticles(Domain* domain, DomainDecompBase* domainDecomp) {
-	domainDecomp->collCommInit(domain->getglobalNumMolecules() * 10);
-
-	for (unsigned i = 1; i < domain->getglobalNumMolecules() + 1; i++) {
-		domainDecomp->collCommAppendUnsLong(this->_cid[i]);
-		domainDecomp->collCommAppendUnsLong(this->_molID[i]);
-		domainDecomp->collCommAppendUnsLong(this->_doODF[i]);
-		domainDecomp->collCommAppendDouble(this->_r1[0][i]);
-		domainDecomp->collCommAppendDouble(this->_r1[1][i]);
-		domainDecomp->collCommAppendDouble(this->_r1[2][i]);
-		domainDecomp->collCommAppendDouble(this->_Q1[0][i]);
-		domainDecomp->collCommAppendDouble(this->_Q1[1][i]);
-		domainDecomp->collCommAppendDouble(this->_Q1[2][i]);
-		domainDecomp->collCommAppendDouble(this->_Q1[3][i]);
-	}
-	domainDecomp->collCommAllreduceSum();
-	for (unsigned i = 1; i < domain->getglobalNumMolecules() + 1; i++) {
-		this->_cid[i] = domainDecomp->collCommGetUnsLong();
-		this->_molID[i] = domainDecomp->collCommGetUnsLong();
-		this->_doODF[i] = domainDecomp->collCommGetUnsLong();
-		this->_r1[0][i] = domainDecomp->collCommGetDouble();
-		this->_r1[1][i] = domainDecomp->collCommGetDouble();
-		this->_r1[2][i] = domainDecomp->collCommGetDouble();
-		this->_Q1[0][i] = domainDecomp->collCommGetDouble();
-		this->_Q1[1][i] = domainDecomp->collCommGetDouble();
-		this->_Q1[2][i] = domainDecomp->collCommGetDouble();
-		this->_Q1[3][i] = domainDecomp->collCommGetDouble();
+void ODF::calculateOrientation(const array<double, 3>& simBoxSize, const Molecule& mol1, const Molecule& mol2,
+							   const array<double, 3>& upVec1) {
+	if (mol2.numDipoles() != 1) {
+		return;
 	}
-	domainDecomp->collCommFinalize();
-	for (unsigned i = 1; i < domain->getglobalNumMolecules() + 1; i++) {
-		this->_upVec1[0][i] = 2 * (this->_Q1[1][i] * this->_Q1[3][i] + this->_Q1[0][i] * this->_Q1[2][i]);
-		this->_upVec1[1][i] = 2 * (this->_Q1[2][i] * this->_Q1[3][i] - this->_Q1[0][i] * this->_Q1[1][i]);
-		this->_upVec1[2][i] = 1 - 2 * (this->_Q1[1][i] * this->_Q1[1][i] + this->_Q1[2][i] * this->_Q1[2][i]);
+
+	// TODO Implement rotation matrices to calculate orientations for dipole direction unit vectors other than [0 0 1];
+
+	// FIXME: rename (almost all of) these to something more meaningful
```

Comment:
@JoshuaMarx address this please

Reply:
Do you mean the TODO part? Currently, for the plugin to calculate the orientations properly, the user must choose [0 0 1] as the standard direction vector for the dipole in the xml file. The standard dipole direction vector is mostly arbitrary and only serves as a reference point for the orientation quaternions. I want the plugin to eventually work with any chosen standard dipole vector but it is not a priority right now since it doesn't affect functionality.

I am renaming some of the variables and writing additional comments =)

Reply:
I only meant the renaming. Thx :)

Reply:
Done!